### DISCUSSION TITLE
Significant increase in sys time in lz4 v1.10.0 compared to lz4 v1.8.3

### DISCUSSION THREAD

**Discussion by sasam** [lz4/lz4#1512] [General]
I have an application (bash & gawk)  that starts 24 threads in parallel that decompress files with lz4 and do some grep in them. The server I run it on is Linux with 48 cores. I have lz4 v1.8.3 installed on that Linux.

Today I compiled lz4 v1.10.0 so I tested how both work.

I noticed that with the new version the thing works faster but that it consumes significantly more sys tyme, so I'm wondering if anyone else has noticed that ?

these are the measurements i got with time of my application with new & older lz4 version
```
lz4 v1.8.3
real 4m13.243s
user 59m14.857s
sys 19m5.086s

lz4 v1.10.0
real 3m46.428s
user 46m16.230s
sys 31m4.964s
```

**Comment by Cyan4973** [comment]
`lz4` `v1.10` uses multi-threading by default, which is inherently heavier for the system than single-thread `v1.8.3`.
To reduce the difference, you might want to compile `lz4` with multi-threading disabled.

**Comment by sasam** [comment]
I compiled with HAVE_MULTITHREAD=0
These are the results of all the versions I tried
```
*** LZ4 command line interface 64-bits v1.8.3, by Yann Collet ***
real    4m35.514s
user    65m41.015s
sys     35m37.523s

real    4m19.986s
user    59m39.904s
sys     40m6.895s

real    4m25.727s
user    65m7.161s
sys     34m58.301s

*** lz4 v1.10.0 64-bit single-thread, by Yann Collet ***
real    3m58.717s
user    51m15.672s
sys     38m3.291s

real    3m42.504s
user    50m9.123s
sys     27m57.677s

real    3m57.439s
user    48m3.638s
sys     32m41.034s

*** lz4 v1.10.0 64-bit multithread, by Yann Collet ***
real    3m36.566s
user    53m27.482s
sys     58m9.895s

real    3m43.637s
user    55m49.879s
sys     60m8.891s

real    3m38.839s
user    51m27.483s
sys     59m24.073s
```
